private_sharing/models.py

```diff
@@ -236,6 +236,7 @@ class DataRequestProject(models.Model):
         ),
         verbose_name="Are you requesting Open Humans usernames?",
     )
+    data_types = models.ManyToManyField("ProjectOntology", related_name="data_types")
```

Comment:
it's unclear why you'd want a manytomany to ProjectOntology which, in turn, has manytomany to Ontology. AFAICT there's just one target: DataType, and each project may have zero, one, or more DataTypes it's registered.

This field is ambiguously named - is it the datatypes the project is registering for allowed uploads? Or is it replacing requested_sources for the requested data access?